MekHQ/unittests/mekhq/campaign/market/ContractMarketAtBGenerationTests.java

```diff
@@ -0,0 +1,1741 @@
+/*
+ * Copyright (c) 2020 - The MegaMek Team. All rights reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.campaign.market;
+
+import mekhq.campaign.Campaign;
+import mekhq.campaign.CampaignOptions;
+import mekhq.campaign.CurrentLocation;
+import mekhq.campaign.Hangar;
+import mekhq.campaign.JumpPath;
+import mekhq.campaign.finances.Accountant;
+import mekhq.campaign.finances.Money;
+import mekhq.campaign.force.Force;
+import mekhq.campaign.mission.AtBContract;
+import mekhq.campaign.rating.IUnitRating;
+import mekhq.campaign.rating.UnitRatingMethod;
+import mekhq.campaign.universe.Faction;
+import mekhq.campaign.universe.FactionHints;
+import mekhq.campaign.universe.Factions;
+import mekhq.campaign.universe.RandomFactionGenerator;
+import mekhq.campaign.universe.PlanetarySystem;
+import mekhq.campaign.universe.Systems;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.Vector;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(Parameterized.class)
+public class ContractMarketAtBGenerationTests {
+
+    private final int gameYear;
+    private final int unitRating;
+    private final boolean isClanEnemy;
+
+    public ContractMarketAtBGenerationTests(int gameYear, int unitRating, boolean isClanEnemy) {
+        this.gameYear = gameYear;
+        this.unitRating = unitRating;
+        this.isClanEnemy = isClanEnemy;
+    }
+
+    @Parameters(name = "Run {index}: gameYear={0}, unitRating={1}, isClanEnemy={2}")
+    public static Iterable<Object[]> data() throws Throwable {
+        List<Integer> gameYears = Arrays.asList(new Integer[] { 2750, 3025, 3055, 3067, 3120 });
+        
+        List<Object[]> parameters = new ArrayList<>();
+        for (int gameYear : gameYears) {
+            for (int rating = IUnitRating.DRAGOON_F; rating <= IUnitRating.DRAGOON_ASTAR; ++rating) {
```

Comment:
Why ++rating instead of rating++?

Reply:
No known corner cases for pre-increment in any of the languages I use (as opposed to post-increment).